lib/bolt/cli.rb

```diff
@@ -516,23 +514,36 @@ def outputter
       @outputter ||= Bolt::Outputter.for_format(@config[:format])
     end
 
-    # Runs a block in a PAL script compiler configured for Bolt.
-    # Catches exceptions thrown by the block and re-raises them as Bolt::CLIError.
+    # Runs a block in a PAL script compiler configured for Bolt.  Catches
+    # exceptions thrown by the block and re-raises them ensuring they are
+    # Bolt::Errors since the script compiler block will squash all exceptions.
     def in_bolt_compiler(opts = [])
       Puppet.initialize_settings(opts)
       r = Puppet::Pal.in_tmp_environment('bolt', modulepath: [BOLTLIB_PATH] + @config[:modulepath], facts: {}) do |pal|
         pal.with_script_compiler do |compiler|
           begin
             yield compiler
           rescue Puppet::PreformattedError => err
-            err.cause
+            # Puppet sometimes rescues exceptions notes the location and reraises
+            # For now return the original error.
+            if err.cause
+              if err.is_a? Bolt::Error
```

Comment:
We throw Bolt::Errors that include a cause?

Reply:
Actually, this is specifically catching `Puppet::PreformattedError`, how would that also be a `Bolt::Error`?

Reply:
oh thats a mistake. it's supposed to be err.cause.is_a?
We do throw Bolt::Errors with causes though

Reply:
Is the line below also supposed to be `err.cause`?